internal/index/manager/manager.go

```diff
@@ -1675,6 +1693,81 @@ func (mgr *Manager) startMonitoringConverters(watcher *fsnotify.Watcher) {
 	}
 }
 
+func (mgr *Manager) startMonitoringPcaps(watcher *fsnotify.Watcher) {
+	go func() {
+		var (
+			waitFor = 500 * time.Millisecond
+			mu      sync.Mutex
+			timers  = make(map[string]*time.Timer)
+		)
+		for {
+			select {
+			case err, ok := <-watcher.Errors:
+				if !ok {
+					return
+				}
+				log.Println("error:", err)
+
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return
+				}
+				log.Println("event:", event)
+
+				if !(event.Has(fsnotify.Create|fsnotify.Write|fsnotify.Chmod) && filepath.Ext(event.Name) == ".pcap") {
+					continue
+				}
+
+				mu.Lock()
+				timer, ok := timers[event.Name]
+				mu.Unlock()
+
+				if !ok {
+					timer = time.AfterFunc(math.MaxInt64, func() {
+						fileInfo, err := os.Stat(event.Name)
+						if err != nil || fileInfo.IsDir() {
+							return
+						}
+
+						src, err := os.Open(event.Name)
+						if err != nil {
+							log.Printf("error while opening new pcap: %v", err)
+							return
+						}
+						defer src.Close()
+						dst, err := os.Create(filepath.Join(mgr.PcapDir, fileInfo.Name()))
+						if err != nil {
+							log.Printf("error while copying new pcap to PcapDir: %v", err)
+							return
+						}
+						defer dst.Close()
+
+						if _, err := io.Copy(dst, src); err != nil {
+							log.Printf("error while copying new pcap to PcapDir: %v", err)
+							return
```

Comment:
The newly created pcap copy should be deleted when the content wasn't copied correctly to not leave empty files behind.

https://github.com/spq/pkappa2/blob/ba1bc3987475b7cb946b130bde8896ba77e5cf10/cmd/pkappa2/main.go#L170-L172